### PR TITLE
Update Braket parallel demo to measure all qubits

### PR DESCRIPTION
**Pull request by antalszava**
**Context**

Braket may raise the following error:
```
botocore.errorfactory.ValidationException: An error occurred (ValidationException) when calling the CreateQuantumTask operation: Device requires all qubits in the program to be measured. This may be caused by declaring non-contiguous qubits or measuring partial qubits
```

This is being resolved in Braket. To make the related demo future proof, updates the demonstration to measure all qubits.

### CODE REVIEW

File: demonstrations/braket-parallel-gradients.py [hunk 1]
```diff
@@ -140,7 +140,10 @@ def circuit(params):
         qml.RX(params[i], wires=i)
     for i in range(n_wires):
         qml.CNOT(wires=[i, (i + 1) % n_wires])
-    return qml.expval(qml.PauliZ(n_wires - 1))
+
+    # Measure all qubits to make sure all's good with Braket
+    observables = [qml.PauliZ(n_wires - 1)] + [qml.Identity(i) for i in range(n_wires - 1)]
+    return qml.expval(qml.operation.Tensor(*observables))
```
Comment (AlbertMitjans) [inline demonstrations/braket-parallel-gradients.py:146]: Could we use `Prod` instead?

```suggestion
    return qml.expval(qml.prod(*observables))
```

**Review by AlbertMitjans — COMMENTED** [review]
I cannot test this because I don't have an AWS account.

Are there any Xanadu credentials that I could use? Or should I create my own personal account?

File: demonstrations/braket-parallel-gradients.py (hunk 1 above)
Comment (timmysilv) [reply demonstrations/braket-parallel-gradients.py:146]: tried running with this change locally and got:
```python
TypeError: Unsupported observable: PauliZ(wires=[24]) @ Identity(wires=[0]) @ Identity(wires=[1]) @ Identity(wires=[2]) @ Identity(wires=[3]) @ Identity(wires=[4]) @ Identity(wires=[5]) @ Identity(wires=[6]) @ Identity(wires=[7]) @ Identity(wires=[8]) @ Identity(wires=[9]) @ Identity(wires=[10]) @ Identity(wires=[11]) @ Identity(wires=[12]) @ Identity(wires=[13]) @ Identity(wires=[14]) @ Identity(wires=[15]) @ Identity(wires=[16]) @ Identity(wires=[17]) @ Identity(wires=[18]) @ Identity(wires=[19]) @ Identity(wires=[20]) @ Identity(wires=[21]) @ Identity(wires=[22]) @ Identity(wires=[23])
```

ran in a python repl and got:
```python
>>> observables = [qml.PauliZ(n_wires - 1)] + [qml.Identity(i) for i in range(n_wires - 1)]
>>> qml.expval(qml.prod(*observables))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/matthews/src/github.com/PennyLaneAI/pennylane/pennylane/measurements.py", line 382, in __repr__
    if self.obs.return_type is None:
AttributeError: 'Prod' object has no attribute 'return_type'
```

**Review by timmysilv — APPROVED** [review]
Ran locally without the change, got the error; with the change, succeeded!

**Pull request merged**